internal/controller/dnsrecord_controller.go

```diff
@@ -152,19 +152,30 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, err
 	}
 
-	return r.updateStatus(ctx, previous, dnsRecord, requeueAfter)
+	return r.updateStatus(ctx, previous, dnsRecord, requeueAfter, nil)
 }
 
-func (r *DNSRecordReconciler) updateStatus(ctx context.Context, previous, current *v1alpha1.DNSRecord, requeueAfter time.Duration) (reconcile.Result, error) {
-	current.Status.QueuedFor = metav1.NewTime(reconcileStart.Add(requeueAfter))
+func (r *DNSRecordReconciler) updateStatus(ctx context.Context, previous, current *v1alpha1.DNSRecord, requeueAfter time.Duration, specErr error) (reconcile.Result, error) {
+	if specErr != nil {
+		current.Status.WriteCounter = previous.Status.WriteCounter
+		current.Status.ValidFor = previous.Status.ValidFor
+		current.Status.QueuedAt = previous.Status.QueuedAt
+		current.Status.QueuedFor = previous.Status.QueuedFor
+	}
```

Comment:
If we get an error ensure that these status fileds are not updated from there previous values